rubin_sim/maf/stackers/date_stackers.py

```diff
@@ -36,6 +38,79 @@ def _run(self, sim_data, cols_present=False):
         return sim_data
 
 
+class ObservationStartTimestampStacker(BaseStacker):
+    """Add the observation start time as a pandas.Timestamp."""
+
+    cols_added = ["start_timestamp"]
+
+    def __init__(
+        self,
+        mjd_col="observationStartMJD",
+    ):
+        self.mjd_col = mjd_col
+        self.cols_req = [self.mjd_col]
+        self.units = [None]
+        self.cols_added_dtypes = ["O"]
+
+    def _run(self, sim_data, cols_present=False):
+        if cols_present:
+            # Column already present in data; assume it is correct and does not
+            # need recalculating.
+            return sim_data
+
+        if len(sim_data[self.mjd_col]) > 0:
+            empty_series = pd.to_datetime(
+                sim_data[self.mjd_col] + 2400000.5, origin="julian", unit="D", utc=True
+            )
+        else:
+            # If we are passed an empty series, be sure to return add an empty
+            # series of the correct type back.
+            # This is handy if the result is being passed to bokeh
+            # for plotting.
+            empty_series = pd.to_datetime(2460000.5, origin="julian", unit="D", utc=True)
+
+        match sim_data[self.mjd_col]:
+            case pd.Series():
+                sim_data["start_timestamp"] = empty_series
+            case np.ndarray():
+                sim_data["start_timestamp"] = empty_series.values
+            case _:
+                sim_data["start_timestamp"] = empty_series.values
+
+        return sim_data
+
+    def run(self, sim_data, override=False):
```

Comment:
I don't understand what you're doing here. 
The "run" method should call "_run", to actually add the data, but I don't see that here. 
It seems to me that this method to add the columns could override _add_columns in the base class instead?

Reply:
My initial attempt at this was to override _add_columns, but that doesn't work, because the dtype I'm trying to add is a pandas thing, not a numpy thing, _add_columns only deals with numpy dtypes, and even if I added it as an "object" type in numpy, if I convert it back to pandas it still gets the dtype wrong. The simplest solution I could come up with is bypassing the _add_columns and _run mechanics entirely.

Reply:
Does this actually add the values though? I saw you had a "_run" method, but I don't see it being called from "run". 
I could just be confused -- if you're not actually calling "_run", does it need to be here?

Reply:
I think this was a mistake on my part -- I'll get rid of it. It's a leftover from when I was attempting to make this work with _run, before I gave up and just did everything in run.